msal-python-conceptual/advanced/managed-identity.md

```diff
@@ -0,0 +1,110 @@
+
+## Examples
+
+In both system- and user-assigned identities, developers need to use <xref:msal.managed_identity.ManagedIdentityClient> to access managed identities.
+
+### System-assigned managed identities
+
+System-assigned managed identities can be used by instantiating <xref:msal.managed_identity.SystemAssignedManagedIdentity> and passing to <xref:msal.managed_identity.ManagedIdentityClient>.
+
+>[!NOTE]
+>You need to include a `http_client` reference, which can be set to `requests.Session()`. This enables MSAL to maintain a pool of connections to the IMDS endpoint.
+
+You can specify the target resource scope when calling [`acquire_token_for_client`](xref:msal.managed_identity.ManagedIdentityClient.acquire_token_for_client).
+
+```python
+import msal
+import requests
+
+managed_identity = msal.SystemAssignedManagedIdentity()
+
+global_app = msal.ManagedIdentityClient(managed_identity, http_client=requests.Session())
+
+result = global_app.acquire_token_for_client(resource='https://vault.azure.net')
+
+if "access_token" in result:
+    print("Token obtained!")
+```
+
+>[!IMPORTANT]
+>You need to enable a system-assigned identity for the resource where the Python code runs; otherwise, no token will be returned.
+
+### User-assigned managed identities
+
+User-assigned managed identities can be used by instantiating <xref:msal.managed_identity.UserAssignedManagedIdentity> and passing to <xref:msal.managed_identity.ManagedIdentityClient>. You will need to specify the **one of the following**:
+
+- Client ID (`client_id`)
+- Resource ID (`resource_id`)
+- Object ID (`object_id`)
+
+>[!NOTE]
+>You need to include a `http_client` reference, which can be set to `requests.Session()`. This enables MSAL to maintain a pool of connections to the IMDS endpoint.
+
+You can specify the target resource scope when calling [`acquire_token_for_client`](xref:msal.managed_identity.ManagedIdentityClient.acquire_token_for_client).
+
+```python
+import msal
+import requests
+
+managed_identity = msal.UserAssignedManagedIdentity(client_id='YOUR_CLIENT_ID')
+
+global_app = msal.ManagedIdentityClient(managed_identity, http_client=requests.Session())
+
+result = global_app.acquire_token_for_client(resource='https://vault.azure.net')
+
+if "access_token" in result:
+    print("Token obtained!")
+```
+
+>[!IMPORTANT]
+>You need to attach a user-assigned identity for the resource where the Python code runs; otherwise, no token will be returned. If an incorrect identifier is used for the user-assigned managed identity, no token will be returned as well.
+
+## Caching
+
+By default, MSAL Python supports in-memory caching. MSAL does not support cache extensibility for managed identity because of security concerns when using distributed cache. Since a token acquired for managed identity belongs to an Azure resource, using a distributed cache might expose it to the other Azure resources sharing the cache.
```

Comment:
No, in MSAL Python there is legit need to extend the token cache for persistence.

```suggestion
By default, MSAL Python supports in-memory caching. MSAL Python also support cache extensibility for managed identity, so that you may persist the token cache on disk. This can be useful if you are writing a command-line script in Python. We do not recommend sharing managed identity token cache among multiple machines.
```

Also, I believe, sharing token cache across multiple VMs using same UAMI would technically be OK. It is just that there will be no economic gain to do so. Fetching a token from a remote (distributed) token cache would require a network call, which is not faster than fetching a new MI token from the current environment and then cache it locally.